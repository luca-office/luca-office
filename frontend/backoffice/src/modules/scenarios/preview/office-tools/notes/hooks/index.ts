export * from "./use-notes-state"
export * from "./use-notes-state-actions"
export * from "./use-notes-survey-events"
