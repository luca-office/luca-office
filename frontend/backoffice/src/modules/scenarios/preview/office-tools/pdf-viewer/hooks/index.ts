export * from "./use-pdf-viewer-state"
export * from "./use-pdf-viewer-survey-events"
