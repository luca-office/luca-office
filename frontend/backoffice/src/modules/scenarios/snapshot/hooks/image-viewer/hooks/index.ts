export * from "./use-image-viewer-snapshot-state"
export * from "./use-image-viewer-snapshot-survey-events"
