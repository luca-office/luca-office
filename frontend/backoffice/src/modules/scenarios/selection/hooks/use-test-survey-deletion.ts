import {useDeleteSurvey, useSurveys} from "shared/graphql/hooks"

export interface UseTestSurveyDeletionHook {
  readonly deleteTestSurveys: () => Promise<void[]>
}

export const useTestSurveyDeletion = (projectId: UUID): UseTestSurveyDeletionHook => {
  const {surveys} = useSurveys(projectId)
  const {deleteEntity: deleteSurvey} = useDeleteSurvey(projectId)

  const deleteTestSurveys = () => {
    return Promise.all(surveys.filter(survey => survey.isTestSurvey).map(({id}) => deleteSurvey(id)))
  }

  return {
    deleteTestSurveys
  }
}
