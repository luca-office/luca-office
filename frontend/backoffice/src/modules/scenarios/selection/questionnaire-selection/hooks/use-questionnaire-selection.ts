import React, {Dispatch, SetStateAction} from "react"
import {useDispatch} from "react-redux"
import {useCreateScenarioQuestionnaire, useQuestionnaires, useScenarioQuestionnaires} from "shared/graphql/hooks"
import {Questionnaire} from "shared/models"
import {first, Option} from "shared/utils"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {Route} from "../../../../../routes"

export interface UseQuestionnaireSelectionHook {
  readonly loading: boolean
  readonly handleBackNavigation: () => void
  readonly eventQuestionnairePreviewForModal: Option<Questionnaire>
  readonly setEventQuestionnairePreviewForModal: Dispatch<SetStateAction<Option<Questionnaire>>>
  readonly questionnaires: Questionnaire[]
  readonly assignedQuestionnaires: Questionnaire[]
  readonly updateAssignment: (selectedEvents: Questionnaire[]) => void
}

export const useQuestionnaireSelection = (
  scenarioId: UUID,
  isRuntimeSurvey: boolean
): UseQuestionnaireSelectionHook => {
  const dispatch = useDispatch()

  const {questionnaires, questionnairesLoading} = useQuestionnaires(isRuntimeSurvey)
  const {scenarioQuestionnaires, scenarioQuestionnairesLoading} = useScenarioQuestionnaires(scenarioId)
  const {createScenarioQuestionnaire} = useCreateScenarioQuestionnaire()

  const [eventQuestionnairePreviewForModal, setEventQuestionnairePreviewForModal] = React.useState<
    Option<Questionnaire>
  >(Option.none())

  const handleBackNavigation = () => dispatch(navigateToRouteAction(Route.ScenarioQuestionnaireDetail, {scenarioId}))

  const createAssignment = (selectedQuestionnaires: Questionnaire[]) => {
    selectedQuestionnaires.map(questionnaire =>
      createScenarioQuestionnaire({
        scenarioId,
        activationDelayInSeconds: 0,
        questionnaireId: questionnaire.id
      })
    )

    first(selectedQuestionnaires).forEach(questionnaire =>
      dispatch(
        navigateToRouteAction(Route.ScenarioQuestionnaireDetail, {scenarioId, questionnaireId: questionnaire.id})
      )
    )
  }

  return {
    loading: questionnairesLoading || scenarioQuestionnairesLoading,
    eventQuestionnairePreviewForModal,
    setEventQuestionnairePreviewForModal,
    questionnaires: questionnaires.getOrElse([]).filter(questionnaire => !!questionnaire.publishedAt),
    handleBackNavigation,
    updateAssignment: createAssignment,
    assignedQuestionnaires: scenarioQuestionnaires.getOrElse([]).map(assigned => assigned.questionnaire)
  }
}
