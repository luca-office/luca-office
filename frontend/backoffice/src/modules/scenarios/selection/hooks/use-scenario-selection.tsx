import * as React from "react"
import {useDispatch} from "react-redux"
import {ProjectModuleType} from "shared/graphql/generated/globalTypes"
import {useCreateProjectModule, useProject, useProjectModules, useScenarios} from "shared/graphql/hooks"
import {Scenario, ScenarioLight} from "shared/models"
import {useCheckUserClaims} from "../../../../hooks/use-check-user-claims"
import {navigateToRouteAction} from "../../../../redux/actions/navigation-action"
import {Route} from "../../../../routes"
import {checkIsProjectFinalized} from "../../../projects/utils"
import {useTestSurveyDeletion} from "./use-test-survey-deletion"

export interface UseScenarioOverviewHook {
  readonly alreadyAssignedScenarios: Scenario[]
  readonly isProjectFinalized: boolean // finalized == existing survey
  readonly openProjectDetail: () => void
  readonly openSelectionDetail: (scenarioId: UUID) => void
  readonly saveScenarioAssignment: (selectedScenarios: ScenarioLight[]) => void
  readonly scenarios: ScenarioLight[]
  readonly userMayFinalizeWithoutPublishing: boolean
}

export const useScenarioSelection = (projectId: UUID): UseScenarioOverviewHook => {
  const dispatch = useDispatch()

  const {scenarios: allScenarios} = useScenarios()
  const {createProjectModule} = useCreateProjectModule()
  const {projectModules, projectModulesLoading} = useProjectModules(projectId)
  const {project, projectLoading} = useProject(projectId)
  const {userClaims, userClaimsCheckLoading} = useCheckUserClaims()
  const {deleteTestSurveys} = useTestSurveyDeletion(projectId)

  const projectScenarios = React.useMemo(
    () =>
      projectModules.reduce(
        (accumulator, module) => (!module.scenario ? accumulator : [...accumulator, module.scenario]),
        [] as Scenario[]
      ),
    [projectModulesLoading]
  )
  const isProjectFinalized = !projectLoading && checkIsProjectFinalized(project)
  const userMayFinalizeWithoutPublishing = !userClaimsCheckLoading && userClaims.mayFinalizeWithoutPublishing

  const openProjectDetail = () => dispatch(navigateToRouteAction(Route.ProjectDetail, {id: projectId}))
  const openSelectionDetail = (scenarioId: UUID) =>
    dispatch(navigateToRouteAction(Route.ScenarioSelectionDetail, {id: projectId, scenarioId}))
  const saveScenarioAssignment = (selectedScenarios: ScenarioLight[]) => {
    const selectedScenarioIds = selectedScenarios.map(scenario => scenario.id)
    const createdScenarioIds = selectedScenarioIds.filter(id => !projectScenarios.some(scenario => scenario.id === id))

    deleteTestSurveys().then(() => {
      createdScenarioIds.forEach(scenarioId =>
        createProjectModule({
          projectId,
          scenarioId,
          moduleType: ProjectModuleType.Scenario
        })
      )
      openProjectDetail()
    })
  }

  return {
    openSelectionDetail,
    alreadyAssignedScenarios: projectScenarios,
    isProjectFinalized,
    scenarios: allScenarios
      .map(scenarios =>
        scenarios.filter(
          scenario =>
            (isProjectFinalized ? !!scenario.publishedAt : true) &&
            (!userMayFinalizeWithoutPublishing ? !scenario.finalizedAt : true)
        )
      )
      .getOrElse([]),
    openProjectDetail,
    saveScenarioAssignment,
    userMayFinalizeWithoutPublishing
  }
}
