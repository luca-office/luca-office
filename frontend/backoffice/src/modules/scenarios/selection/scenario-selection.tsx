import {css} from "@emotion/react"
import * as React from "react"
import {OverviewCard} from "shared/components"
import {ScenarioLight} from "shared/models"
import {headerHeight, spacingHuger, spacingLarge, spacingMedium, subHeaderHeight} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {CardDurationInfo} from "../../../components"
import {ModuleSelectionContainer} from "../../common/module-selection/module-selection-container"
import {useScenarioSelection} from "./hooks/use-scenario-selection"

interface Props {
  readonly projectId: UUID
}

export const ScenarioSelection: React.FC<Props> = ({projectId}) => {
  const {
    alreadyAssignedScenarios,
    isProjectFinalized,
    openProjectDetail,
    openSelectionDetail,
    saveScenarioAssignment,
    scenarios,
    userMayFinalizeWithoutPublishing
  } = useScenarioSelection(projectId)

  const {t} = useLucaTranslation()

  return (
    <ModuleSelectionContainer<ScenarioLight>
      customCardOverviewStyles={styles.cardOverview}
      entities={scenarios}
      multiSelection={true}
      alreadyAssignedEntities={alreadyAssignedScenarios}
      onSelectionConfirmed={saveScenarioAssignment}
      userMayFinalizeWithoutPublishing={userMayFinalizeWithoutPublishing}
      subheaderConfig={{
        entityFilterType: isProjectFinalized ? "scenarioSelectionPublished" : "scenarioSelection",
        labelKey: "projects__scenario_header_label",
        navigationButton: {
          labelKey: "projects__header_details_label",
          onClick: openProjectDetail
        },
        customFilterHeaderStyles: css({gridTemplateColumns: "1fr 1fr 1fr 3fr"})
      }}
      footerConfig={{
        emptySelectionKey: "project_module_selection__empty_selection_scenario",
        entitySelectionKey: "project_module_selection__selected_scenario"
      }}
      renderContent={(scenario, footer) => (
        <OverviewCard
          key={scenario.id}
          onClick={() => openSelectionDetail(scenario.id)}
          headerText={scenario.name}
          headerInfo={<CardDurationInfo maxDurationInSeconds={scenario.maxDurationInSeconds} t={t} />}
          noAnimationOnHover={true}
          text={scenario.description}
          footer={footer}
          tags={scenario.tags}
        />
      )}
      getEditingStatusConfig={scenario => ({isFinalized: !!scenario.finalizedAt, isPublished: !!scenario.publishedAt})}
    />
  )
}

const styles = {
  cardOverview: css({
    maxHeight: `calc(100vh - ${2 * headerHeight + 2 * subHeaderHeight + spacingHuger + spacingLarge + spacingMedium}px)`
  })
}
