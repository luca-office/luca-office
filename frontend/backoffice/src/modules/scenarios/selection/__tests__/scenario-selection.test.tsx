import {MockedProvider} from "@apollo/client/testing"
import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {DetailViewHeader, OverviewCard} from "shared/components"
import {checkLoginMock, projectScenariosMock, projectsMock, scenariosMock} from "shared/graphql/__mocks__"
import {checkLoginQuery} from "shared/graphql/queries"
import {CardOverview} from "../../../../components"
import {ModuleSelectionContainer} from "../../../common/module-selection/module-selection-container"
import * as scenarioOverviewHook from "../hooks/use-scenario-selection"
import {UseScenarioOverviewHook} from "../hooks/use-scenario-selection"
import {ScenarioSelection} from "../scenario-selection"

const testScenarios = [...scenariosMock, {...scenariosMock[0], finalizedAt: null, publishedAt: null}]
const hookValuesDefault: UseScenarioOverviewHook = {
  openSelectionDetail: jest.fn(),
  openProjectDetail: jest.fn(),
  alreadyAssignedScenarios: projectScenariosMock,
  saveScenarioAssignment: jest.fn(),
  scenarios: testScenarios,
  userMayFinalizeWithoutPublishing: true,
  isProjectFinalized: true
}
const projectId = projectsMock[0].id

const stateSpy = jest.spyOn(scenarioOverviewHook, "useScenarioSelection")

const getComponent = () => (
  <MockedProvider
    mocks={[
      {
        request: {
          query: checkLoginQuery
        },
        result: {
          data: {
            checkLogin: checkLoginMock
          }
        }
      }
    ]}>
    <ScenarioSelection projectId={projectId} />
  </MockedProvider>
)

describe("scenario-selection", () => {
  it("renders correctly", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)
    const component = create(getComponent())

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure (0 Scenario)", async () => {
    stateSpy.mockReturnValue({...hookValuesDefault, scenarios: []})

    const component = mount(getComponent())
    const moduleSelectionContainer = component.find(ModuleSelectionContainer)
    expect(moduleSelectionContainer).toHaveLength(1)

    expect(moduleSelectionContainer.find(OverviewCard)).toHaveLength(0)
    expect(moduleSelectionContainer.find(CardOverview)).toHaveLength(1)
    expect(moduleSelectionContainer.find(DetailViewHeader)).toHaveLength(1)
  })
  it("has correct structure (2 finalized, published Scenario)", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)

    const component = mount(getComponent())
    const moduleSelectionContainer = component.find(ModuleSelectionContainer)
    expect(moduleSelectionContainer).toHaveLength(1)

    expect(moduleSelectionContainer.find(OverviewCard)).toHaveLength(
      testScenarios.filter(scenario => !!scenario.finalizedAt || !scenario.publishedAt).length + 1
    )
    expect(moduleSelectionContainer.find(DetailViewHeader)).toHaveLength(1)
  })
  it("has correct structure (test survey)", async () => {
    stateSpy.mockReturnValue({...hookValuesDefault, isProjectFinalized: false})

    const component = mount(getComponent())
    const moduleSelectionContainer = component.find(ModuleSelectionContainer)
    expect(moduleSelectionContainer).toHaveLength(1)
    expect(moduleSelectionContainer.find(OverviewCard)).toHaveLength(testScenarios.length)
    expect(moduleSelectionContainer.find(DetailViewHeader)).toHaveLength(1)
  })
})
