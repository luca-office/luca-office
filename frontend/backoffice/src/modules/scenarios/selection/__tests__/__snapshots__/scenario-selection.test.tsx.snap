// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenario-selection renders correctly 1`] = `
<div
  className="content css-usun20-Content"
>
  <div
    className="css-7pb0dh-DetailViewHeader"
  >
    <button
      className="css-rmd9d0-Button"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="css-95j2hm-Button"
      >
        <div
          className="css-1lrzij5-Icon"
          dangerouslySetInnerHTML={
            {
              "__html": {},
            }
          }
        />
        <div
          className="css-u7npei-DetailViewHeader"
        >
          projects__header_details_label
        </div>
      </div>
    </button>
    <div
      className="css-tqo6e0-DetailViewHeader"
    >
      projects__scenario_header_label
    </div>
  </div>
  <div
    className="content-container css-1v8oche-Content"
  >
    <div
      className="css-1ldr313-SubHeader"
    >
      <div
        className="css-1vv8qb3"
        data-testid="custom-select"
      >
        <div
          className="css-0"
        >
          <div
            className="css-13s2sfk-Heading"
          >
            subheader__typ
          </div>
        </div>
        <div
          className="css-18vooac"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="custom-select css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-1l51fcv-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-1fdsijx-ValueContainer"
              >
                <div
                  className="single-value css-1ry7giw-singleValue"
                >
                  subheader__filter_by_all_entities
                </div>
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-readonly={true}
                  className="css-mohuvp-dummyInput-DummyInput"
                  disabled={false}
                  id="react-select-2-input"
                  inputMode="none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  tabIndex={0}
                  value=""
                />
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1uei4ir-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <div
                    className="css-l0jg45-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1vv8qb3"
        data-testid="custom-select"
      >
        <div
          className="css-0"
        >
          <div
            className="css-13s2sfk-Heading"
          >
            subheader__sort
          </div>
        </div>
        <div
          className="css-18vooac"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="custom-select css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-1l51fcv-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-1fdsijx-ValueContainer"
              >
                <div
                  className="single-value css-1ry7giw-singleValue"
                >
                  subheader__sort_by_creation_date
                </div>
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-readonly={true}
                  className="css-mohuvp-dummyInput-DummyInput"
                  disabled={false}
                  id="react-select-3-input"
                  inputMode="none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  tabIndex={0}
                  value=""
                />
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1uei4ir-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <div
                    className="css-l0jg45-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-17as3x0"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="css-13s2sfk-Heading"
        >
          subheader__search
        </div>
        <div
          className="css-1unfqpo"
        >
          <input
            className="css-dga44z"
            defaultValue=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="subheader__search_placeholder"
            type="text"
          />
          <div
            className="css-v3avn8-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="css-1w1rgjg-CardOverview"
    >
      <section
        className="css-153l9ug-CardOverview"
      >
        <div
          className="css-1xzym25-Card"
          onClick={[Function]}
        >
          <div>
            <div
              className="card-header css-1x5pyem-CardHeader"
            >
              <div
                className="css-1b415o5-OverviewCard"
              >
                <div
                  className="css-1i6jmor-Heading"
                >
                  Scenario 1
                </div>
              </div>
              <div
                className="header-time css-adnwg9-CardDurationInfo"
                data-testid="card-duration-info"
              >
                <div
                  className="css-1f6p2vj-Heading"
                >
                  90 unit__minutes_short
                </div>
                <div
                  className="css-kgdmdb-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
            </div>
            <div>
              <div
                className="css-1v230f2-Text"
              >
                description
              </div>
            </div>
          </div>
          <div>
            <div
              className="css-1gj5t7m-TagsOnCards"
            >
              <div
                className="css-1ws8uc4-Tag"
              >
                <div
                  className="css-12ia3h2-Text"
                >
                  placeholder__tags
                </div>
              </div>
            </div>
            <div
              className="card-footer css-awvo71-CardFooter"
              onClick={[Function]}
            >
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  03.08.2020
                </div>
              </div>
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  John Doe
                </div>
              </div>
              <div
                className="editing-status-indicator css-z3bxxz-EditingStatusIndicator"
              >
                <div
                  className="css-l0jg45-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-1cywa3f-Button"
                  disabled={false}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-mj8vo1-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                placeholder__add_to_selection
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1xzym25-Card"
          onClick={[Function]}
        >
          <div>
            <div
              className="card-header css-1x5pyem-CardHeader"
            >
              <div
                className="css-1b415o5-OverviewCard"
              >
                <div
                  className="css-1i6jmor-Heading"
                >
                  Scenario 2
                </div>
              </div>
            </div>
            <div>
              <div
                className="css-1v230f2-Text"
              >
                description
              </div>
            </div>
          </div>
          <div>
            <div
              className="css-1gj5t7m-TagsOnCards"
            >
              <div
                className="css-1ws8uc4-Tag"
              >
                <div
                  className="css-12ia3h2-Text"
                >
                  placeholder__tags
                </div>
              </div>
            </div>
            <div
              className="card-footer css-awvo71-CardFooter"
              onClick={[Function]}
            >
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  03.08.2020
                </div>
              </div>
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  John Doe
                </div>
              </div>
              <div
                className="editing-status-indicator css-z3bxxz-EditingStatusIndicator"
              >
                <div
                  className="css-l0jg45-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-1cywa3f-Button"
                  disabled={false}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-mj8vo1-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                placeholder__add_to_selection
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1xzym25-Card"
          onClick={[Function]}
        >
          <div>
            <div
              className="card-header css-1x5pyem-CardHeader"
            >
              <div
                className="css-1b415o5-OverviewCard"
              >
                <div
                  className="css-1i6jmor-Heading"
                >
                  Scenario 2
                </div>
              </div>
            </div>
            <div>
              <div
                className="css-1v230f2-Text"
              >
                description
              </div>
            </div>
          </div>
          <div>
            <div
              className="css-1gj5t7m-TagsOnCards"
            >
              <div
                className="css-1ws8uc4-Tag"
              >
                <div
                  className="css-12ia3h2-Text"
                >
                  placeholder__tags
                </div>
              </div>
            </div>
            <div
              className="card-footer css-awvo71-CardFooter"
              onClick={[Function]}
            >
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  03.08.2020
                </div>
              </div>
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  John Doe
                </div>
              </div>
              <div
                className="editing-status-indicator css-z3bxxz-EditingStatusIndicator"
              >
                <div
                  className="css-l0jg45-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-1cywa3f-Button"
                  disabled={false}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-mj8vo1-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                placeholder__add_to_selection
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1xzym25-Card"
          onClick={[Function]}
        >
          <div>
            <div
              className="card-header css-1x5pyem-CardHeader"
            >
              <div
                className="css-1b415o5-OverviewCard"
              >
                <div
                  className="css-1i6jmor-Heading"
                >
                  Scenario 2
                </div>
              </div>
            </div>
            <div>
              <div
                className="css-1v230f2-Text"
              >
                description
              </div>
            </div>
          </div>
          <div>
            <div
              className="css-1gj5t7m-TagsOnCards"
            >
              <div
                className="css-1ws8uc4-Tag"
              >
                <div
                  className="css-12ia3h2-Text"
                >
                  placeholder__tags
                </div>
              </div>
            </div>
            <div
              className="card-footer css-awvo71-CardFooter"
              onClick={[Function]}
            >
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  03.08.2020
                </div>
              </div>
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  John Doe
                </div>
              </div>
              <div
                className="editing-status-indicator css-z3bxxz-EditingStatusIndicator"
              >
                <div
                  className="css-l0jg45-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-1cywa3f-Button"
                  disabled={false}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-mj8vo1-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                placeholder__add_to_selection
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1xzym25-Card"
          onClick={[Function]}
        >
          <div>
            <div
              className="card-header css-1x5pyem-CardHeader"
            >
              <div
                className="css-1b415o5-OverviewCard"
              >
                <div
                  className="css-1i6jmor-Heading"
                >
                  Scenario 1
                </div>
              </div>
              <div
                className="header-time css-adnwg9-CardDurationInfo"
                data-testid="card-duration-info"
              >
                <div
                  className="css-1f6p2vj-Heading"
                >
                  90 unit__minutes_short
                </div>
                <div
                  className="css-kgdmdb-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
            </div>
            <div>
              <div
                className="css-1v230f2-Text"
              >
                description
              </div>
            </div>
          </div>
          <div>
            <div
              className="css-1gj5t7m-TagsOnCards"
            >
              <div
                className="css-1ws8uc4-Tag"
              >
                <div
                  className="css-12ia3h2-Text"
                >
                  placeholder__tags
                </div>
              </div>
            </div>
            <div
              className="card-footer css-awvo71-CardFooter"
              onClick={[Function]}
            >
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  03.08.2020
                </div>
              </div>
              <div
                className="css-1839z0z-CardFooterItem"
              >
                <div
                  className="css-1lrzij5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1de4iva-Text"
                >
                  John Doe
                </div>
              </div>
              <div
                className="editing-status-indicator css-z3bxxz-EditingStatusIndicator"
              >
                <div
                  className="css-l0jg45-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-1cywa3f-Button"
                  disabled={false}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-mj8vo1-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                placeholder__add_to_selection
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
  <div
    className="bottom-action-bar css-1004l4i-ContentBottomActionbar"
  >
    <div
      className="card-footer css-30zwfm-CardFooter"
    >
      <div
        className="css-1jbe7ev-Text"
      >
        project_module_selection__empty_selection_scenario
      </div>
      <button
        className="css-ncq1no-Button"
        disabled={true}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          reference_books__scenario_add_selected_label
        </div>
      </button>
    </div>
  </div>
</div>
`;
